.github/workflows/build_and_test.yml

```diff
@@ -369,150 +375,166 @@ jobs:
               }
             }"
 
-  docker-image:
-    runs-on: [ self-hosted, Linux, k8s-runner ]
-    needs: [ pg_regress-tests, other-tests ]
-    if: |
-      (github.ref_name == 'main' || github.ref_name == 'release') &&
-      github.event_name != 'workflow_dispatch'
+  dockerfile-check:
+    name: 'dockerfile check'
+    if: github.event_name != 'workflow_dispatch'
+    runs-on: dev
+    container: 369495373322.dkr.ecr.eu-central-1.amazonaws.com/base:latest
     outputs:
-      build-tag: ${{steps.build-tag.outputs.tag}}
+      value: ${{ steps.dockerfile-check.outputs.any_changed }}
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+
+      - name: Get specific changed files
+        id: dockerfile-check
+        uses: tj-actions/changed-files@v24.1
+        with:
+          files: |
+            Dockerfile
+            Dockerfile.compute-tools
+            ./vendor/postgres/Dockerfile
+
+  build-neon-image:
+    name: 'kaniko-neon'
+    # force building for now
+    if: needs.dockerfile-check.outputs.value != 'true'
+    runs-on: dev
+    needs: [ dockerfile-check ]
+    container: gcr.io/kaniko-project/executor:debug
+    environment: dev
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
```

Comment:
Do we use `@v1` for a particular reason? I think on other jobs we use `@v3`?

Reply:
For v3 you need node, which didn't work for this image

Reply:
Ahhh, okay, worth adding a comment then.
Also we should use a tag for this image as well

Reply:
Kaniko is strict on forward & backwards compatibility, similar to aws-cli I won't pin this.

Reply:
> Kaniko is strict on forward & backwards compatibility, similar to aws-cli I won't pin this.

Ok, can we pin a major version at least? Using just `debug` makes it unclear what is running (and why `debug`?)

Reply:
Debug is needed else there is no shell to configure the ECR login, pinned on most recent release